evalml/tests/model_understanding_tests/test_permutation_importance.py

```diff
@@ -0,0 +1,272 @@
+    component_graph = ['Drop Columns Transformer', OneHotEncoder, DateTimeFeaturizer, 'Random Forest Classifier']
+
+
+class LinearPipelineWithImputer(BinaryClassificationPipeline):
+    component_graph = ['Imputer', OneHotEncoder, DateTimeFeaturizer, 'Random Forest Classifier']
+
+
+class LinearPipelineSameFeatureUsedByTwoComponents(BinaryClassificationPipeline):
+    component_graph = ['Imputer', DateTimeFeaturizer, OneHotEncoder, 'Random Forest Classifier']
+
+
+class LinearPipelineTwoEncoders(BinaryClassificationPipeline):
+    component_graph = ['Imputer', DateTimeFeaturizer, OneHotEncoder, OneHotEncoder, "Random Forest Classifier"]
+
+
+class LinearPipelineWithTextFeatures(BinaryClassificationPipeline):
+    component_graph = ['Imputer', 'Drop Columns Transformer', TextFeaturizer, OneHotEncoder, 'Random Forest Classifier']
+
+
+class LinearPipelineWithDoubling(BinaryClassificationPipeline):
+    component_graph = ['Select Columns Transformer', DoubleColumns, DoubleColumns, DoubleColumns, 'Random Forest Classifier']
+
+
+class LinearPipelineWithTargetEncoderAndOHE(BinaryClassificationPipeline):
+    component_graph = ['Imputer', DateTimeFeaturizer, OneHotEncoder, 'Target Encoder', "Random Forest Classifier"]
+
+
+class LinearPipelineCreateFeatureThenDropIt(BinaryClassificationPipeline):
+    component_graph = ['Select Columns Transformer', DoubleColumns, 'Drop Columns Transformer', 'Random Forest Classifier']
+
+
+class DagTwoEncoders(BinaryClassificationPipeline):
+    component_graph = {
+        'Imputer': ['Imputer'],
+        'SelectNumeric': ["Select Columns Transformer", "Imputer"],
+        'SelectCategorical1': ["Select Columns Transformer", "Imputer"],
+        'SelectCategorical2': ["Select Columns Transformer", "Imputer"],
+        'OHE_1': ['One Hot Encoder', 'SelectCategorical1'],
+        'OHE_2': ['One Hot Encoder', 'SelectCategorical2'],
+        'DT': ['DateTime Featurization Component', "SelectNumeric"],
+        'Estimator': ['Random Forest Classifier', 'DT', 'OHE_1', 'OHE_2'],
+    }
+
+
+class DagReuseFeatures(BinaryClassificationPipeline):
+    component_graph = {
+        'Imputer': ['Imputer'],
+        'SelectNumeric': ["Select Columns Transformer", "Imputer"],
+        'SelectCategorical1': ["Select Columns Transformer", "Imputer"],
+        'SelectCategorical2': ["Select Columns Transformer", "Imputer"],
+        'OHE_1': ['One Hot Encoder', 'SelectCategorical1'],
+        'OHE_2': ['One Hot Encoder', 'SelectCategorical2'],
+        'DT': ['DateTime Featurization Component', "SelectNumeric"],
+        'OHE_3': ['One Hot Encoder', 'DT'],
+        'Estimator': ['Random Forest Classifier', 'OHE_3', 'OHE_1', 'OHE_2'],
+    }
+
+
+test_cases = [(LinearPipelineWithDropCols, {"Drop Columns Transformer": {'columns': ['country']}}),
+              (LinearPipelineWithImputer, {}),
+              (LinearPipelineSameFeatureUsedByTwoComponents, {'DateTime Featurization Component': {'encode_as_categories': True}}),
```

Comment:
Amazing. I was wondering what happens when the datetime featurizer creates categorical columns that the OHE then uses, since it'd rely on the output cols of the datetime featurizer as provenance. 🤩